browser/ui/ai_chat/ai_chat_brave_search_throttle_delegate_impl.cc

```diff
@@ -0,0 +1,25 @@
+/* Copyright (c) 2024 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/browser/ui/ai_chat/ai_chat_brave_search_throttle_delegate_impl.h"
+
+#include "chrome/browser/ui/browser_window/public/browser_window_features.h"
+#include "chrome/browser/ui/browser_window/public/browser_window_interface.h"
+#include "chrome/browser/ui/tabs/public/tab_interface.h"
+#include "chrome/browser/ui/views/side_panel/side_panel_coordinator.h"
+#include "chrome/browser/ui/views/side_panel/side_panel_entry_id.h"
+
+namespace ai_chat {
+
+void AIChatBraveSearchThrottleDelegateImpl::OpenLeo(
+    content::WebContents* web_contents) {
+  auto* tab = tabs::TabInterface::GetFromContents(web_contents);
```

Comment:
I think we should recheck the allowlist here or even put all the logic here and make it `MaybeOpenLeo` and then have `OpenLeoImpl` which is the virtual method implemented by the platform specific delegate? I'd like to make sure it can't be called without checking the allowlist.

Reply:
so basically I'm proposing that the base class implements `MaybeOpenLeo` and calls a protected `OpenLeoImpl` virtual method